docs/metrics/latest

```diff
@@ -533,6 +533,21 @@ etcd_server_is_leader
 # type: "counter"
 etcd_server_leader_changes_seen_total
 
+# name: "etcd_server_is_learner"
+# description: "Whether or not this member is a learner. 1 if is, 0 otherwise."
+# type: "gauge"
+etcd_server_is_learner
+
+# name: "etcd_debugging_learner_promote_failures"
+# description: "The total number of learner promote  failures (likely learner not ready)."
```

Comment:
So this counter will only get updated if the local member becomes leader, right? Consider a user sending promote requests to a server which is not the leader at the moment. Then the user checks metrics, and sees 0 promote failures and 0 promote success. Feels like the description needs to be clearer to avoid user confusion. Maybe 'The total number of learner promotion failures while the server is leader'?

Also there are 3 possible reasons that a promote could fail, maybe we should not suggest it is likely to be one of them?

Reply:
I get your point. user can use prometheus sum query to get the total number of the failures. As a debugging metrics I think it is fine add failure reasons as the metrics label. @jingyih

Reply:
OK.